tests/integration/rdb.tcl

```diff
@@ -170,6 +170,70 @@ start_server {} {
         }
         assert_equal [s rdb_changes_since_last_save] 0
     }
+
+    test {bgsave cancel aborts save} {
+        r config set save ""
+        # Generating RDB will take some 100 seconds
+        r config set rdb-key-save-delay 1000000
+        populate 100 "" 16
+
+        r bgsave
+        wait_for_condition 50 100 {
+            [s rdb_bgsave_in_progress] == 1
+        } else {
+            fail "bgsave not done"
+        }
+        set fork_child_pid [get_child_pid 0]
+        
+        assert {[r bgsave cancel] eq {Background bgsave cancelled}}
+        set temp_rdb [file join [lindex [r config get dir] 1] temp-${fork_child_pid}.rdb]
+        # Temp rdb must be deleted
+        wait_for_condition 50 100 {
+            ![file exists $temp_rdb]
+        } else {
+            fail "bgsave temp file was not deleted after cancel"
+        }
+    }
+
+    test {bgsave cancel response with error when no ongoing save} {
+        r config set save ""
+        # Generating RDB will take some 100 seconds
```

Comment:
Copy-paste? This testcase doesn't use an ongoing save.

Reply:
Yeh. I think I will merge this to the previous test